### ISSUE TITLE
Bug inside Diffusers repo

### ISSUE THREAD

**Issue by brandnewx** [TheLastBen/fast-stable-diffusion#815]
Originally, the dreambooth training noise scheduler is:

```
    noise_scheduler = DDPMScheduler(
        beta_start=0.00085, beta_end=0.012, beta_schedule="scaled_linear", num_train_timesteps=1000
    ) 
```
Which was later incorrectly changed to:
```
noise_scheduler = DDPMScheduler.from_config(args.pretrained_model_name_or_path, subfolder="scheduler")
```
Why is it a bug?

Because the noise scheduler in the pretrained models is used for inference only. If you use that for dreambooth training, it will mess up everything. In fact, some models now have DDIMScheduler for inference. If you load the DDIM config into a DDPMScheduler for training, it will produce wrong results.

**Comment by TheLastBen** [comment]
I changed it to match the official diffusers update, with this change, and if you check the scheduler,  and the V2-768 has the v_prediction parameter , if I keep the initial line, the 768 will not train at all.

**Comment by brandnewx** [comment]
I believe the correct approach would be using the previous code for SD v1.x training and using the updated noise_scheduler code for SD v2.x training.

Apart from that, if training SD v1.x, we need to force `epsilon` prediction type too by always setting `target = noise` without checking the condition.

**Comment by brandnewx** [comment]
After several experiments to get back the exact pre-merge results (because DDPMScheduler now forces `epsilon `prediction by default). Previously, DDPMScheduler used `sample` prediction. So need to add `prediction_type="sample"` during the init.

```
    noise_scheduler = DDPMScheduler(
        beta_start=0.00085, beta_end=0.012, beta_schedule="scaled_linear", num_train_timesteps=1000, **prediction_type="sample"**
    )
```

**Comment by TheLastBen** [comment]
try a comparison with and without specifying the prediction_type, even though I think if not specified, it will use the default type

**Comment by brandnewx** [comment]
Before the recent merge, the default prediction_type was "sample" but after the merge, it's now "epsilon". I don't think epsilon prediction works with dreambooth training,

**Comment by TheLastBen** [comment]
I'll check it out

**Comment by TheLastBen** [comment]
> Previously,

when ? did prediction_type="sample" exist before in the script ?

**Comment by brandnewx** [comment]
> > Previously,
> 
> when ? did prediction_type="sample" exist before in the script ?

It's in the HF original DDPMScheduler.py script. Before the merge, the default prediction was sample. Recently, HF team modified DDPMScheduler.py to support `v-prediction`, so they changed the default to epsilon.

**Comment by TheLastBen** [comment]
this is where it was changed, check the training script : https://github.com/huggingface/diffusers/commit/c18941b01ad0ea6b07d020f353d81153c632a374

**Comment by brandnewx** [comment]
I believe Diffusers is broken for SD 1.x dreambooth since then. Now even after all the reverts to old code, the image results are still horrible. I'm going to revert the Diffusers to the pre-merge.

**Comment by TheLastBen** [comment]
are you using the A100 or the standard T4 ?

**Comment by brandnewx** [comment]
I train on T4.

Today is a nasty day. HuggingFace updated the Transformers and their other libraries to enforce strict safety checker, so even the pre-merge code stopped working. I guess this is one of the reasons why you prefetched those pip packages into a zip.
